2022/02/index.js

```diff
@@ -2,13 +2,93 @@ const { promises: fs } = require('fs');
 
 async function getInput() {
   const input = (await fs.readFile(`${__dirname}/input.txt`))
-    .toString();
+    .toString()
+    .trim();
+
+  return input
+    .split('\n')
+    .map(round => round.split(' '));
 }
 
 async function p1() {
+  const selectionScore = {
+    'X': 1, // rock
+    'Y': 2, // paper
+    'Z': 3, // scissors
+  };
+
+  const opponentToYourMove = {
+    'A': 'X', // rock
+    'B': 'Y', // paper
+    'C': 'Z', // scissors
+  }
+
+  const input = await getInput();
+  return input.reduce(
+    (acc, [opponentMove, myMove]) => {
+      let totalScore = acc + selectionScore[myMove];
+
+      if(opponentToYourMove[opponentMove] === myMove) {
+        return totalScore + 3;
+      }
+
+      if(
+        (opponentMove === 'A' && myMove === 'Y') ||
+        (opponentMove === 'B' && myMove === 'Z') ||
+        (opponentMove === 'C' && myMove === 'X')
+      ) {
+        return totalScore + 6;
+      }
+
+      return totalScore;
+    },
+    0
+  );
 }
 
 async function p2() {
+  const selectionScore = {
+    'A': 1, // rock
+    'B': 2, // paper
+    'C': 3, // scissors
+  };
+
+  const loseMap = {
+    'A': 'C', // rock - scissors
+    'B': 'A', // paper - rock
+    'C': 'B', // scissors - paper
```

Comment:
These maps are a pretty neat idea tbh

Reply:
Didn't think about it until I saw the mess of all the I statements I used to have haha

Reply:
I converted my chars to numbers, and then checked if they were bigger or smaller